routing/dht/routing.go

```diff
@@ -170,7 +168,7 @@ func (dht *IpfsDHT) FindProviders(ctx context.Context, key u.Key) ([]peer.PeerIn
 // to the given key
 func (dht *IpfsDHT) getClosestPeers(ctx context.Context, key u.Key) (<-chan peer.ID, error) {
 	e := log.EventBegin(ctx, "getClosestPeers", &key)
-	tablepeers := dht.routingTable.NearestPeers(kb.ConvertKey(key), AlphaValue)
+	tablepeers := dht.routingTable.ListPeers()
```

Comment:
From the kademlia paper:

"The lookup initiator starts by picking (alpha) nodes from its closest non-empty k-bucket (or if that bucket is empty, it takes the alpha closest nodes its knows of). The initiator then sends parallel, asynchronous FIND_NODE RPCs to the alpha nodes it has chosen"

So, im not saying we have to follow the spec, but i dont think using all the peers in the routing table is the right thing to do.